### DISCUSSION TITLE
Unique class names

### DISCUSSION THREAD

**Discussion by Oseryx** [adobe/react-spectrum#4844] [Q&A] [answered]
Hello, I've been reading the react spectrum code to learn more about UI libraries, so there's a thing I fail to understand how it's done, what library/code is used to achieve that

In the code I clearly see (for example): `.spectrum-Slider` but then in the browser it becomes `.spectrum-Slider_4ea6b` I know what happened but I don't understand how it happened

Thanks in advance

**Comment by reidbarber** [comment, marked as answer]
This is done by [CSS Modules](https://github.com/css-modules/css-modules) during the build process. It creates unique class names that are locally scoped so that you don't have to worry about classes with the same name in different files overwriting each other if they're on the same page. Here are docs on how to use it in Parcel: https://parceljs.org/languages/css/#css-modules

**Reply by Oseryx** [reply]
Thanks for the info, I was looking for this exactly
Have a great day!
